recipes/ignition-tools/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import conan.tools.files
+
+required_conan_version = ">=1.29.1"
+
+
+class IgnitionUitlsConan(ConanFile):
+    name = "ignition-tools"
+    license = "Apache-2.0"
+    homepage = "https://ignitionrobotics.org/libs/tools"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Provides general purpose classes and functions designed for robotic applications.."
+    topics = ("ignition", "robotics", "tools")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake", "cmake_find_package_multi"
+    exports_sources = "CMakeLists.txt", "patches/**"
+    _cmake = None
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn(
+                "{} recipe lacks information about the {} compiler support.".format(
+                    self.name, self.settings.compiler
+                )
+            )
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires c++17 support. The current compiler {} {} does not support it.".format(
+                        self.name,
+                        self.settings.compiler,
+                        self.settings.compiler.version,
+                    )
+                )
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version][str(self.settings.os)], destination=self._source_subfolder, strip_root=True)
```

Comment:
```suggestion
        tools.get(**self.conan_data["sources"],destination=self._source_subfolder, strip_root=True)
```

Sorry, OS is only valid for installers, my bad.